src/internal_rules.jl

```diff
@@ -558,3 +558,418 @@ function EnzymeRules.reverse(
+        fact::Annotation{C}, B::Union{Const};
+        kwargs...
+) where {C <: Cholesky}
+    retval = copy(B.val)
+    ldiv!(fact.val, retval)
+    return retval
+end
+
+function EnzymeRules.forward(
+        func::Const{typeof(\)},
+        RT::Type{<:Union{Const, DuplicatedNoNeed, Duplicated}},
+        fact::Annotation{C}, B::Union{Const, Duplicated};
+        kwargs...
+) where {C <: Cholesky}
+    retval = copy(B.val)
+    ldiv!(fact.val, retval)
+    retdval = []
+    if isa(fact, Duplicated) && isa(B, Duplicated)
+        retdval = zeros(length(retval))
+        mul!(retdval, fact.dval.U, retval)
+        mul!(retdval, fact.dval.L, retdval)
+        retdval .= B.dval .- retdval
+        ldiv!(fact.val, retdval)
+    elseif isa(fact, Duplicated) && isa(B, Const)
+        retdval = zeros(length(retval))
+        mul!(retdval, A, B.val)
+        mul!(retdval, -1, retdval)
+        ldiv!(fact.val, retdval)
+    elseif isa(fact, Const) && isa(B, Duplicated)
+        retdval = copy(B.dval)
+        ldiv!(fact.val, retdval)
+    elseif isa(fact, Const) && isa(B, Const)
+        nothing
+    else
+        error("Error in forward $(typeof(func)) Enzyme rule $(typeof(fact)) $(typeof(B)).")
+    end
+    if RT <: Const
+        return retval
+    elseif (RT <: DuplicatedNoNeed) || (RT <: BatchDuplicatedNoNeed)
+        return retdval
+    else
+        return Duplicated(retval, retdval)
+    end
+end
+
+function EnzymeRules.forward(
+        func::Const{typeof(\)},
+        RT::Type{<:Union{Const, BatchDuplicatedNoNeed, BatchDuplicated}},
+        fact::Annotation{C}, B::Union{Const, BatchDuplicated{T,N}};
+        kwargs...
+) where {C <: Cholesky, T <: AbstractArray{<:AbstractFloat}, N}
+    retval = copy(B.val)
+    ldiv!(fact.val, retval)
+    bretdval = ntuple(Val(N)) do b
+        if isa(fact, BatchDuplicated) && isa(B, BatchDuplicated)
+            retdval = zeros(length(retval))
+            mul!(retdval, fact.dval[b].U, retval)
+            mul!(retdval, fact.dval[b].L, retdval)
+            retdval .= B.dval[b] .- retdval
+            ldiv!(fact.val, retdval)
+        elseif isa(fact, BatchDuplicated) && isa(B, Const)
+            retdval = zeros(length(retval))
+            mul!(retdval, A, B.val)
+            mul!(retdval, -1, retdval)
+            ldiv!(fact.val, retdval)
+        elseif isa(fact, Const) && isa(B, BatchDuplicated)
+            retdval = copy(B.dval)
+            ldiv!(fact.val, retdval)
+        elseif isa(fact, Const) && isa(B, Const)
+            nothing
+        else
+            error("Error in forward \\ Enzyme rule $(typeof(fact)) $(typeof(B)).")
+        end
+        retdval
+    end
+    if RT <: Const
+        return retval
+    elseif (RT <: DuplicatedNoNeed) || (RT <: BatchDuplicatedNoNeed)
+        return retdval
+    else
+        return BatchDuplicated(retval, ntuple(i -> bretdval[i], Val(N)))
+    end
+end
+
+function EnzymeRules.forward(
+        func::Const{typeof(ldiv!)},
```

Comment:
This rule seems wrong (and unnecssary) remove it?

Reply:
It's because of ambiguity with the batch and nonbatched methods both can be called when arguments are `Const`.